contracts/marketplace/Marketplace.sol

```diff
@@ -103,44 +97,51 @@ contract Marketplace is Ownable, Pausable, Destructible {
   /**
     * @dev Cancel an already published order
     * @param assetId - ID of the published NFT
+    * @param nftAddress - Non fungible registry address
     * @param priceInWei - Price in Wei for the supported coin.
-    * @param expiresAt - Duration of the auction (in hours)
+    * @param expiresAt - Duration of the order (in hours)
     */
-  function createOrder(uint256 assetId, uint256 priceInWei, uint256 expiresAt) public whenNotPaused {
-    address assetOwner = nonFungibleRegistry.ownerOf(assetId);
+  function createOrder(uint256 assetId, address nftAddress, uint256 priceInWei, uint256 expiresAt) public whenNotPaused {
+    require(isContract(nftAddress));
+
+    ERC721 nftRegistry = ERC721(nftAddress);
+    address assetOwner = nftRegistry.ownerOf(assetId);
+
     require(msg.sender == assetOwner);
-    require(nonFungibleRegistry.isAuthorized(address(this), assetId));
+    require(nftRegistry.isApprovedForAll(assetOwner, address(this)));
     require(priceInWei > 0);
     require(expiresAt > block.timestamp.add(1 minutes));
 
-    bytes32 auctionId = keccak256(
+    bytes32 orderId = keccak256(
```

Comment:
Should we include the NFT Registry address here to be better sure of not having a hash collision?